common/src/main/java/org/dash/wallet/common/data/NetworkDataProvider.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2022 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.dash.wallet.common.data
+
+/**
+ * Dash Network data provider
+ *
+ */
+
+interface NetworkDataProvider {
+    fun getMasternodeAPY(): Double
+}
```

Comment:
The advantage of this interface is that the caller need not know about the wallet or dashj.  

Because the APY is to be displayed on ExploreFragment, which is in the explore module, we can use this and avoid using dashj or crowdnode related code.

Reply:
Perhaps this class can be merged with BlockchainStateProvider that was added to Coinbase.

The Masternode APY does depend on the current state of the blockchain, though the BlockchainState object does not contain all the information we need to determine the APY.

Reply:
I was thinking the same thing. Seem like a related functionality